dags/hindawi/hindawi_api_client.py

```diff
@@ -29,8 +33,30 @@ def get_articles_metadata(self, parameters, doi=None):
                 break
             return response_content
 
-    # TODO: don't forget to download pdf when it will be needed
+    def get_file(self, parameters, extension, doi=None):
+        path_segments_for_files = ["journals", "ahep"]
+        try:
+            doi_part = doi.replace("10.1155/", "") + f".{extension}"
+            request = Request(
+                base_url=self.files_url,
+                path_segments=path_segments_for_files + [doi_part],
+                parameters=parameters,
+            )
+            return request.get_response_bytes()
+        except AttributeError as e:
+            self.logger.error(f"No doi was passed for getting the file. Error: {e}")
+        except RequestException as e:
+            self.logger.error(f"Request failed witt exception: {e}")
+
+    def get_pdf_file(self, parameters, doi=None):
+        return self.get_file(parameters, "pdf", doi)
```

Comment:
The responsibility logic of the pdf file's path should be in this function and not in the `get_file`. The same applies to `pdfa` and `xml`.